### ISSUE TITLE
Can't install Sentry - Composer error

### ISSUE THREAD

**Issue by heartbeatLV** [getsentry/sentry-laravel#284]
- Laravel Version: 5.8
- PHP Version: 7.2.19

### Description:
I can't install the package through composer.

```
Required package "sentry/sentry" is not installed: cannot detect its version
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
Installation failed, reverting ./composer.json to its original content.
```

**Comment by stayallive** [comment]
What command do you run to install Sentry and are you able to share your full `composer.json` so I can try to reproduce?

**Comment by heartbeatLV** [comment]
from the docs
```
composer require sentry/sentry-laravel:1.3.1
```

**Comment by stayallive** [comment]
Can you share your full `composer.json` so I can try to reproduce?

**Comment by heartbeatLV** [comment]
```
{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "require": {
        "php": "^7.1.3",
        "barryvdh/laravel-cors": "^0.11.3",
        "fideloper/proxy": "^4.0",
        "laravel/framework": "5.8.*",
        "laravel/tinker": "^1.0",
        "tymon/jwt-auth": "^1.0"
    },
    "require-dev": {
        "beyondcode/laravel-dump-server": "^1.0",
        "filp/whoops": "^2.0",
        "fzaninotto/faker": "^1.4",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^3.0",
        "nunomaduro/phpinsights": "^1.7",
        "phpunit/phpunit": "^7.5"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/"
        },
        "classmap": [
            "database/seeds",
            "database/factories"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]
    }
}
```

**Comment by stayallive** [comment]
Could you try removing your `composer.lock` file and running `composer install` and than try again running `composer require sentry/sentry-laravel:1.3.1`?

I think something somehow got messed up, it seems to work for me with your `composer.json`. I'm a bit puzzled why this is an issue.

**Comment by heartbeatLV** [comment]
I deleted `composer.lock`, than I ran the command `composer install` and than `composer require sentry/sentry-laravel:1.3.1`
result: same error

I made another attempt:
I deleted `composer.lock` and `vendor` folder, than I ran the command `composer clearcache` and  `composer install` and than `composer require sentry/sentry-laravel:1.3.1`
result: same error

![immagine](https://user-images.githubusercontent.com/33943143/66898812-2b5eb400-effa-11e9-8c74-e5e33e64d205.png)

**Comment by stayallive** [comment]
@Jean85 do you have some things to try and figure it out, I can't replicate the issue but it does look like it comes from the package version detection and I know we had some problems with that before.

---

@heartbeatLV it is correct `sentry/sdk` has no download since that is a package that is only there to include some other packages. It's a so-called meta package and has no code: https://github.com/getsentry/sentry-php-sdk#sentry-pre-configured-php-sdk

**Comment by Jean85** [comment]
The issue is probably the same as https://github.com/getsentry/sentry-laravel/issues/274#issuecomment-536901862

You cannot reproduce because you probably have the `package-versions` already installed; it won't probably work if you remove that. It's a race condition between post-install hooks. I will try to report this upstream.

**Comment by jur3** [comment]
The issue was in Ocramius/PackageVersions package.
They fixed issue in 1.5.0, so upgrade  Ocramius/PackageVersions to 1.5.0

**Comment by Jean85** [comment]
Oh so the issue is probably solved by https://github.com/Ocramius/PackageVersions/pull/82. Thanks for reporting back on this.

Closing as solved, please reply further if that doesn't solve this issue. Since that feature is in a tagged version that requires PHP 7.3, we could open a PR to backport it.

**Comment by Jean85** [comment]
I've backported https://github.com/Ocramius/PackageVersions/pull/82 with https://github.com/Ocramius/PackageVersions/pull/112, and it got released as [1.4.1](https://github.com/Ocramius/PackageVersions/releases/tag/1.4.1), which is compatible with PHP 7.1. 

This issue should no longer be present.
